modules/redisearch.ts

```diff
@@ -111,84 +116,93 @@ export class Redisearch extends Module {
                 args = args.concat([
                     'GEOFILTER',
                     parameters.geoFilter.field,
-                    parameters.geoFilter.lon.toString(),
-                    parameters.geoFilter.lat.toString(),
-                    parameters.geoFilter.radius.toString(),
+                    `${parameters.geoFilter.lon}`,
+                    `${parameters.geoFilter.lat}`,
+                    `${parameters.geoFilter.radius}`,
                     parameters.geoFilter.measurement
                 ])
             if(parameters.inKeys !== undefined)
-                args = args.concat(['INKEYS', parameters.inKeys.num.toString(), parameters.inKeys.field])
+                args = args.concat(['INKEYS', `${parameters.inKeys.length}`].concat(parameters.inKeys.map(inKeysItem => `${inKeysItem}`)))
             if(parameters.inFields !== undefined)
-                args = args.concat(['INFIELDS', parameters.inFields.num.toString(), parameters.inFields.field])
+                args = args.concat(['INFIELDS', `${parameters.inFields.length}`].concat(parameters.inFields.map(inFieldItem => `${inFieldItem}`)))
             if(parameters.return !== undefined) {
-                args.push('RETURN');
-                if(parameters.return.num)
-                    args.push(parameters.return.num.toString());
-                if(parameters.return.fields)    
-                    parameters.return.fields.forEach(field => {
-                        args.push(field.name)
-                        if(field.as)
-                            args = args.concat(['AS', field.as])
-                    })
+                args.push('RETURN')
+                let itemCount = 0
+                let tempList = []
+                for (const returnItem of parameters.return) {
+                    if(typeof returnItem === "string") {
+                        //Strings can be pushed directly
+                        tempList.push(returnItem)
+                        itemCount++
+                    } else {
+                        //Objects need to be converted to strings
+                        tempList.push(returnItem.field)
+                        itemCount++
+                        if(returnItem.as !== undefined) {
+                            tempList = tempList.concat(["AS", returnItem.as])
+                            itemCount += 2
+                        }
+                    }
+                }
+                args.push(`${itemCount}`)
+                args = args.concat(tempList)
             }
             if(parameters.summarize !== undefined) {
                 args.push('SUMMARIZE')
                 if(parameters.summarize.fields !== undefined) {
                     args.push('FIELDS')
-                    for(const field of parameters.summarize.fields) {
-                        args = args.concat([field.num.toString(), field.field]);
-                    }
+                    args.push(`${parameters.summarize.fields.length}`)
```

Comment:
Same thing like before, seems like I accidentally moved the num key inside the fields objects instead of outside of it.
You can make it optional to pass the num, else it will use the length. I want people to be able to negatively use the SDK as well. Needs to be transparent. Please use this in more places you see.